x-pack/plugins/ml/public/application/datavisualizer/file_based/components/results_links/results_links.tsx

```diff
@@ -89,7 +124,7 @@ export const ResultsLinks: FC<Props> = ({
               />
             }
             description=""
-            href={`${basePath.get()}/app/discover#/?&_a=(index:'${indexPatternId}')${globalStateString}`}
+            href={`${basePath.get()}/app/discover/?&_a=(index:'${indexPatternId}')${globalStateString}`}
```

Comment:
please use the URL generator provided by the Discover plugin

Reply:
Updated here https://github.com/elastic/kibana/pull/76735/commits/1efaed29b3be3a6a6a9fa6ae27caf30815bc3070